lib/realtime/block_waiver.ex

```diff
@@ -104,8 +104,8 @@ defmodule Realtime.BlockWaiver do
     {end_time, _} = List.last(trip_stop_time_waivers)
 
     %__MODULE__{
-      start_time: start_time,
-      end_time: end_time,
+      start_time: Util.Time.timestamp_for_time_of_day(start_time),
+      end_time: Util.Time.timestamp_for_time_of_day(end_time),
```

Comment:
`date_of_timestamp` says in its `@doc` "This is the date that roles over at midnight, not the service date". So I think this won't handle the date rollover correctly.

Consider a trip from 02:30 to 03:00 and a trip from 26:30 to 27:00, both of which are running at the same time and both of which get a block waiver. This `timestamp_for_time_of_day` conversion needs different dates in order to turn those different times of days into the same timestamps. If you use the date that roles over at midnight, then the block waiver for 26:30 would get a timestamp 24 hours in the future.

The other place where `timestamp_for_time_of_day` is used (in `ghost.ex`) knows what date to use for the conversion by holding on to the date it found when digging through the calendar looking for active blocks. Unfortunately, since we're not starting from the calendar here, we don't already have a date, so that strategy won't work.

The `next_time_of_day_for_timestamp_after` conversion that's used in `timepoint_status.ex` converts the opposite direction, but uses a strategy that might be useful for this case. It tries doing the naive conversion, but if the time_of_day would be more than an hour before or more than 23 hours after the time we expect to be approximately correct, then we adjust the date so we get something that's in the range we expect.

Since this is the first time we're trying to do a conversion from time_of_day to timestamp that needs to handle the date rollover correctly, I think you'll need to write a new `next_timestamp_for_time_of_day_after` function, and choose a date that gives you a timestamp that's close to `Util.Time.now()`. For block waivers, maybe you'd aim for a timestamp that's between 4 hours ago and 20 hours in the future?

(Side note: there are a few uses of `date_of_timestamp()` in `time.ex` and `data.ex`, but the callers all check if the naive answer is correct and then adjust accordingly.)

Reply:
We actually have the block and know it is active, so I changed it to behave the same way.